logsmanagement/logsmanagement.c

```diff
@@ -0,0 +1,1312 @@
+    }
+};
+
+
+struct File_infos_arr *p_file_infos_arr = NULL;
+
+
+static struct Chart_meta chart_types[] = {
+    {.type = FLB_TAIL,      .init = generic_chart_init,   .update = generic_chart_update},
+    {.type = FLB_WEB_LOG,   .init = web_log_chart_init,   .update = web_log_chart_update},
+    {.type = FLB_KMSG,      .init = kernel_chart_init,    .update = kernel_chart_update},
+    {.type = FLB_SYSTEMD,   .init = systemd_chart_init,   .update = systemd_chart_update},
+    {.type = FLB_DOCKER_EV, .init = docker_ev_chart_init, .update = docker_ev_chart_update},
+    {.type = FLB_SYSLOG,    .init = systemd_chart_init,   .update = systemd_chart_update},
+    {.type = FLB_SERIAL,    .init = generic_chart_init,   .update = generic_chart_update},
+    {.type = FLB_MQTT,      .init = mqtt_chart_init,      .update = mqtt_chart_update}
+};
+
+g_logs_manag_config_t g_logs_manag_config = {
+    .update_every = 1,
+    .update_timeout = UPDATE_TIMEOUT_DEFAULT,
+    .use_log_timestamp = CONFIG_BOOLEAN_AUTO,
+    .circ_buff_max_size_in_mib = CIRCULAR_BUFF_DEFAULT_MAX_SIZE / (1 MiB),
+    .circ_buff_drop_logs = CIRCULAR_BUFF_DEFAULT_DROP_LOGS,
+    .compression_acceleration = COMPRESSION_ACCELERATION_DEFAULT,
+    .db_mode = GLOBAL_DB_MODE_DEFAULT,
+    .disk_space_limit_in_mib = DISK_SPACE_LIMIT_DEFAULT,  
+    .buff_flush_to_db_interval = SAVE_BLOB_TO_DB_DEFAULT,
+    .enable_collected_logs_total = ENABLE_COLLECTED_LOGS_TOTAL_DEFAULT,
+    .enable_collected_logs_rate = ENABLE_COLLECTED_LOGS_RATE_DEFAULT
+};
+
+static flb_srvc_config_t flb_srvc_config = {
+    .flush          = "0.1",
+    .http_listen    = "0.0.0.0",
+    .http_port      = "2020",
+    .http_server    = "false",
+    .log_path       = "NULL",
+    .log_level      = "info"
+};
+
+static logs_manag_db_mode_t db_mode_str_to_db_mode(const char *const db_mode_str){
+    if(!db_mode_str || !*db_mode_str) return g_logs_manag_config.db_mode;
+    else if(!strcasecmp(db_mode_str, "full")) return LOGS_MANAG_DB_MODE_FULL;
+    else if(!strcasecmp(db_mode_str, "none")) return LOGS_MANAG_DB_MODE_NONE;
+    else return g_logs_manag_config.db_mode;
+}
+
+static bool metrics_dict_conflict_cb(const DICTIONARY_ITEM *item __maybe_unused, void *old_value, void *new_value, void *data __maybe_unused){
+    ((metrics_dict_item_t *)old_value)->num += ((metrics_dict_item_t *)new_value)->num;
+    return true;
+}
+
+/** 
+ * @brief Cleanup p_file_info struct
+ * @param p_file_info The struct of File_info type to be cleaned up.
+ * @todo  Pass p_file_info by reference, so that it can be set to NULL. */
+static void p_file_info_destroy(struct File_info *p_file_info){
+
+    // TODO: Clean up rrd / chart stuff.
+
+    if(unlikely(!p_file_info)){
+        collector_info("p_file_info_destroy() called but p_file_info == NULL - already destroyed?");
+        return;
+    }
+
+    char chart_name[100];
+    snprintfz(chart_name, 100, "%s", p_file_info->chart_name ? p_file_info->chart_name : "Unknown");
```

Comment:
Just to beat Thiago here, to suggest `100 - 1` :-)

Reply:
I've noticed the pattern we seem to obey to, but why??

Reply:
@MrZammler you won! :smile: 
@Dim-P as you said, this is a pattern, and to have some code different here it is a breaking pattern, this is the first reason to keep.
The second I will write according a conclusion, but I was not the person that made the pattern. As you know C language always use `NULL` to end a string, and I understand this pattern as a road to force developers to reserve the necessary space and also to think that his array has the enough size to store the information.

Best regards!